macos/FluentUI/Link.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+/// A lightweight hyperlink control
+@objc(MSFLink)
+open class Link : NSButton {
+	
+	/// Initialies a link with a content string that is displayed, and an underlying
+	/// - Parameters:
+	///   - content: The visible text of the link that the user sees.
+	///   - url: The URL that is opened when the link is clicked
+	@objc public init(content: String, url: NSURL) {
+		self.content = content
+		self.url = url
+		super.init(frame: .zero)
+		alignment = .natural
+		isBordered = false
+		target = self
```

Comment:
These are standard button APIs and the fact that we rely on clients not changing them might be an issue. Should we at the least override the setter for target and action with a preconditionFailure(), make them unavailable, or some other way of saying "don't do this, our control relies on it"?

Reply:
Oh! That's a great idea! I've been wondering how to stop clients from changing properties I need constant 😃

Reply:
Few problems with this:
If I just blindly put `preconditionFailure` calls on the didSet of `target`, `action`, etc..., then this would crash trying to set those in init. I can add a private flag `doneInitializing` that we have a `guard` on inside `didSet`, but that feels less clean.

Also, `NSButton` (along with any other alternative class I use) have lots of properties.. should I set them all to crash if you access them? or just the ones that I override, like we mentioned? This problem happens whether I subclass or return a stylized object, so I'm inclined to just clearly document "Please don't mess with X properties" and add `preconditionFailure` calls with the message "Improper use of Link Control, please do not override this property, or subclass/make your own custom object if you wish to use this"

Reply:
Upon thinking this through, I think differently - do we need to restrict clients? If they need to change the alignment or target, we should just let them? As long as the default they get from using fluentui APIs gives them the appearance they want, I don't think we should spend effort trying to protect them and instead should assume they are well aware of AppKit and can hopefully find other properties to perfectly adapt this to their needs?

Reply:
Related - overriding target seems like a very necessary thing if we're providing a default behavior of opening a URL. I could easily see someone asking to have it open their help viewer or settings or something like that and that functionality should already be baked into the control by AppKit